tests/integration/targets/aws_secret/tasks/basic.yml

```diff
@@ -1,112 +1,573 @@
+    vars:
+      _caller_info: '{{ caller_info.arn.split(":") }}'
+      _base_arn: 'arn:{{_caller_info[1]}}:secretsmanager:{{ aws_region }}'
+    set_fact:
+      account_arn: '{{_base_arn}}:{{_caller_info[4]}}:secret:'
 
   # ============================================================
-  # Module parameter testing
+  # Creation testing
   # ============================================================
-  - name: test with no parameters
+  - name: add secret to AWS Secrets Manager
     aws_secret:
+      name: "{{ secret_name }}"
+      state: present
+      secret_type: 'string'
+      secret: "{{ super_secret_string }}"
     register: result
-    ignore_errors: true
-    check_mode: true
 
-  - name: assert failure when called with no parameters
+  - name: assert correct keys are returned
     assert:
       that:
-         - result.failed
-         - 'result.msg.startswith("missing required arguments:")'
+        - result.changed
+        - '"arn" in result.secret'
+        # - '"last_accessed_date" in result.secret'
+        - '"last_changed_date" in result.secret'
+        - '"name" in result.secret'
+        - '"version_ids_to_stages" in result.secret'
+        - result.secret.arn.startswith(account_arn)
+        # ARN includes a random string after the name
+        - secret_name in result.secret.arn
+        - result.secret.name == secret_name
+        - result.secret.version_ids_to_stages | length == 1
 
-  # ============================================================
-  # Creation/Deletion testing
-  # ============================================================
-  - name: add secret to AWS Secrets Manager
+  - name: save the ARN for later comparison
+    set_fact:
+      secret_arn: '{{ result.secret.arn }}'
+
+  - name: no changes to secret
     aws_secret:
       name: "{{ secret_name }}"
       state: present
       secret_type: 'string'
       secret: "{{ super_secret_string }}"
     register: result
 
+  - name: assert correct keys are returned
+    assert:
+      that:
+        - not result.changed
+        - '"arn" in result.secret'
+        - '"created_date" in result.secret'
+        - '"last_accessed_date" in result.secret'
+        - '"last_changed_date" in result.secret'
+        - '"name" in result.secret'
+        - '"version_ids_to_stages" in result.secret'
+        - result.secret.arn == secret_arn
+        - result.secret.name == secret_name
+        - result.secret.version_ids_to_stages | length == 1
+
+  - name: Set secret description
+    aws_secret:
+      name: "{{ secret_name }}"
+      description: 'this is a change to this secret'
+      state: present
+      secret_type: 'string'
+      secret: "{{ super_secret_string }}"
+    register: result
+
   - name: assert correct keys are returned
     assert:
       that:
         - result.changed
-        - result.arn is not none
-        - result.name is not none
-        - result.tags is not none
-        - result.version_ids_to_stages is not none
```

Comment:
This wasn't actually doing what the author thought it was doing.  The values were actually all in `result.secret`